File: c++/Stackproblem.cpp
#include <bits/stdc++.h>
using namespace std;
string isBalanced(string s) {
    stack<char> st;
    for(auto c:s){
        switch (c){
            case '(':
            case '{':
            case '[':
                  st.push(c);
                   break;
            case '}':
                if(st.empty() || st.top()!='{' )
                    return "false";
                st.pop();
                break;
            case ']':
                if(st.empty() || st.top()!='[')
                    return "false";
                st.pop();
                break;
            case ')':
                if(st.empty() || st.top()!='(')
                    return "false";
                st.pop();
                break;
            default: break;
        }
    }
    return st.empty() ? "true":"false";

}
int main(){
   int t;
   cin>>t;
   while(t--){
    string s;
    cin>>s;
    cout<<isBalanced(s)<<endl;
   }
   return 0;
}

File: java/stack_problem.java

import java.util.*;
import java.util.Stack;
public class Main
{
public static boolean isbalanced(String s){
Stack<Character> s1 = new Stack<>();
int c;
for (int i = 0; i < s.length(); i++){
switch(s.charAt(i)){
case '[':
case '{':
case '(':
s1.push(s.charAt(i));
break;
case ']':

if(s1.empty() || (c = Character.compare(s1.peek(),'[')) != 0)  
return false;
s1.pop();
break;
case '}':

if(s1.empty() || (c = Character.compare(s1.peek(),'{')) != 0)
return false;
s1.pop();
break;
case ')':

if(s1.empty() || (c = Character.compare(s1.peek(),'(')) != 0)
return false;
s1.pop();
break;
}
}

if(s1.empty())
return true;
else 
return false;
}
public static void main(String[] args) {
Scanner sc = new Scanner(System.in);

while(sc.hasNextLine()){
String s = sc.nextLine().trim();

System.out.println(isbalanced(s));

}

}
}
